### ISSUE TITLE
kops fails when using c5 instances on aws

### ISSUE THREAD

**Issue by dharmisha** [kubernetes/kops#3868]
I am running kops  version 1.7.1 in aws, i used the amazon/amzn-ami-hvm-2017.09.1.20171103-x86_64-gp2 image, which has ena support. However, when I specify the --node-size c5.large, it fails.

c5 instances are launched recently, however is there a way kops can launch c5 instances?

**Comment by pieterza** [comment]
+1; perhaps kops should not care what size we specify, and let aws tell us if we were wrong?  Else it's always catchup

**Comment by chrislovecnm** [comment]
It is fixed in master and the current beta. Also the Debian base image that kops uses out of the box does not support ena driver. C5 requires that.

So use the beta and you need a different ami

**Comment by justinsb** [comment]
If we can't get this supported in the default image, this at least has to be a warning.

**Comment by adrianchifor** [comment]
I agree, please put a warning, was banging my head against the table all day of why pods were failing randomly on c5.4xlarge nodes.

**Comment by coderanger** [comment]
Copying the specific error here to aid others in googling:

```
Launching a new EC2 instance. Status Reason: Enhanced networking with the Elastic Network Adapter (ENA) is required for the 'c5.large' instance type. Ensure that you are using an AMI that is enabled for ENA. Launching EC2 instance failed.
```

**Comment by chrislovecnm** [comment]
Here is the issue for ENA driver with the base image https://github.com/kubernetes/kube-deploy/pull/292

**Comment by dharmisha** [comment]
It is fixed in master and the current beta. Also the Debian base image that kops uses out of the box does not support ena driver. C5 requires that.So use the beta and you need a different ami
--


I used the beta Version 1.8.0-beta.1 (git-9b71713) binary with amazon/amzn-ami-hvm-2017.09.1.20171103-x86_64-gp2   ami which has support for ena. It still gives an error saying

nodes.spec.machineType: Invalid value: "c5.2xlarge": machine type specified is invalid

Please let me know what needs to be done. This solution is urgently required

**Comment by chrislovecnm** [comment]
https://github.com/kubernetes/kube-deploy/pull/292

Thanks needs to be tested and completed

**Comment by bcorijn** [comment]
Having ENA support by default would be great for all instances, not just C5. I've been running the modified image from my PR for several months now without any issues and was hoping it could get merged before the 1.8 stable release.

**Comment by chrislovecnm** [comment]
@bcorjin the image and the kops releases are not always tied together

**Comment by chrislovecnm** [comment]
#1558 is related, and https://github.com/kubernetes/kube-deploy/pull/292 needs to be fixed in the base image

**Comment by justinsb** [comment]
There are three aspects:

1) recognize c5 instances (done)
2) build a new image supporting ENA (I've merged the supporting fixes to the image and am validating them, so more or less done)
3) support NVME volume mounting in protokube, if we want to run a master on c5.  *not done*  Kubernetes won't support nvme for 1.8, but we're debating whether we can get it into 1.9.  Likely we'll punt this aspect out of the 1.8 milestone, but we'll see...

**Comment by chrislovecnm** [comment]
In regards to number 3.  This means that k8s will not support mount pv as well, I assume.

**Comment by justinsb** [comment]
Exactly - so I have two PRs (with much the same code), one that adds nvme mounting to  protokube, one that adds it to k8s.

**Comment by dharmisha** [comment]
@justinsb @chrislovecnm 
Hi, No new binary has been released which support c5 instances. Is there a specific image that I need to use other than  amazon/amzn-ami-hvm-2017.09.1.20171103-x86_64-gp2  to have kops recognize c5? If not what needs to be done for having it resolved.

**Comment by thuandt** [comment]
kops 1.8.1 has support c5 instances but  EC2 instance will failed. We need AMI update also

**Comment by dharmisha** [comment]
Any idea about the timeline?

**Comment by onyxraven** [comment]
@justinsb mind linking those PRs for easy tracking from this ticket?

**Comment by chrislovecnm** [comment]
This is fixed in 1.8. Please see release notes, let us know if you need more details!!

**Comment by thuandt** [comment]
FYI, if you want use c5 please ensure you didn't use EBS volume. Because kubernetes 1.8 can't mount it as devices path.

Waiting for 1.9 you want to use c5 with EBS

**Comment by chrislovecnm** [comment]
To be clear pvc volumes are not supported in k8s 1.8 - will be supported in 1.9

**Comment by diegocn** [comment]
Is c5 instances supposed to work on 1.9.x now? I've started a cluster with c5.large nodes and the root volumes just got 8GB on / partition (but the volume starts with 128GB) and I got errors with PODs trying to attach PVCs.

**Comment by thuandt** [comment]
@diegocn PVCs is only works with `stretch` image not `jessie` image.

current stable still use `jessie` as i see

**Comment by diegocn** [comment]
Hello @thuandt, thank you.
I'll try to use supported AWS Instances, such as t2.large instead of c5.large for now.

**Comment by chenrui333** [comment]
+1

**Comment by diegocn** [comment]
Running a few POCs with other workloads (not K8S), c5 instances are far better than t2. It would be really nice to have support for these instance type.

**Comment by futurama92** [comment]
Today i try to launch a cluster with c5 instances but the cluster doesn't work...

**Comment by thuandt** [comment]
@faheem-cliqz problem is upgrade. when do `kops upgrade` it will replaced `stretch` with `jessie` :)

**Comment by chenrui333** [comment]
@thuandt do you have any code reference for that?

**Comment by faheem-cliqz** [comment]
@thuandt yup the upgrade part is a problem. We still get the upgrade messages. Just ignoring them for now, and try to upgrade as much as we can with cluster and instance group specs :)

**Comment by bcorijn** [comment]
@chenrui333 upgrade pulls it's information from the channels, which have jessie images. https://github.com/kubernetes/kops/blob/master/channels/stable

**Comment by geojaz** [comment]
closed in #5660
